below/tc/src/types.rs

```diff
@@ -0,0 +1,217 @@
+use netlink_packet_route::tc;
+use netlink_packet_route::tc::{
+    TcAttribute, TcFqCodelXstats, TcMessage, TcOption, TcQdiscFqCodelOption,
+};
+use serde::{Deserialize, Serialize};
+
+const FQ_CODEL: &str = "fq_codel";
+
+/// `Tc` represents a traffic control qdisc.
+#[derive(Default, Clone, PartialEq, Debug, Serialize, Deserialize)]
+pub struct TcStat {
```

Comment:
I think you should include a new `MajMin` struct to uniquely identify the qdisc. I believe the `TcHandle` (maj, min) is all part of the `TcHeader` of `TcMessage`. https://docs.rs/netlink-packet-route/latest/netlink_packet_route/tc/struct.TcMessage.html

Reply:
We already have [handle](https://github.com/facebookincubator/below/pull/8210/files#diff-dacccd72d1d453bb617e574175da08f6712a02732f875df1637e9a86b0d2bb16R17) but from some testing it looks like it does not _uniquely_ identify the qdisc.
From my understanding, a handle would uniquely identify the qdisc for a given interface and a parent. So it's the combination of all these 3 fields (interface, parent, handle), eg:

```swift
TcStat { handle: 0, parent: 4294967295, major: 0, minor: 0, if_index: 1, if_name: "lo", kind: "noqueue", ... }
TcStat { handle: 0, parent: 4294967295, major: 0, minor: 0, if_index: 2, if_name: "ens5", kind: "mq", ... }
TcStat { handle: 0, parent: 2, major: 0, minor: 0, if_index: 2, if_name: "ens5", kind: "fq_codel", ... }
TcStat { handle: 0, parent: 1, major: 0, minor: 0, if_index: 2, if_name: "ens5", kind: "fq_codel", ... }
TcStat { handle: 0, parent: 4294967295, major: 0, minor: 0, if_index: 3, if_name: "docker0", ... }
TcStat { handle: 0, parent: 4294967295, major: 0, minor: 0, if_index: 4, if_name: "br-bddb616e0d48", ... }
TcStat { handle: 0, parent: 4294967295, major: 0, minor: 0, if_index: 76, if_name: "vethdc90bd5", ... }
TcStat { handle: 0, parent: 4294967295, major: 0, minor: 0, if_index: 108, if_name: "veth2dc7d25", ... }
```

Reply:
Yep agreed, we'd need the combination of the 3. I'm happy with it as is, but in the long term, I think we should be keying on the unique identifier.